### DISCUSSION TITLE
emitTo does nothing

### DISCUSSION THREAD

**Discussion by aldesrahim** [filamentphp/filament#2935] [Help] [answered]
### Package

filament/filament

### Package Version

v2.12

### Laravel Version

v9.11

### Livewire Version

v2.6

### PHP Version

PHP 8.1.2

### Bug description

Hello everyone,

So, I have `SaleResource` and within my `index` page I included a widget, and what the widget does is it will get the `queryString` from table's filters and perform some queries to get some data, since I want to pass the date filters from my table to my widget.

So my index page looks like this:
![image](https://user-images.githubusercontent.com/37100012/175861463-6caa79e4-79ef-4aad-bddb-9b26f81701a9.png)

And this is my widget class:
![image](https://user-images.githubusercontent.com/37100012/175861362-2e788fb1-4665-49a0-bca1-5a98b8811b55.png)

My problem is, whenever the table's filters change (`date_from` and `date_until`) and it emits an event to the widget, its just not fired, I have tried to `die n dump` inside the widget and it seems like nothing have changed or fired.

I am very sorry for my bad English, I hope you all can understand what I am trying to deliver,
Thank you for your time.

**Comment by studiowizjo** [comment, marked as answer]
You can emit event this way to widget. I recommend to check if `updateTableFiltersDateDateFrom()` is firing after changing filters. You can put `dd()` there. If so, go further and check if your listener in widget is firing. 

If not, that means the method name is incorrect, so you have to adjust it or change method name to more general `updatedTableFilters()`. 

At the end check if there is something in `$this->tableFilters`. Im not sure if this way data will be there. If not, you can pass it as 3rd argument in `emitTo()`.

Here is working example, how I did it: https://github.com/filamentphp/filament/discussions/2552

It is worth mentioning that currently chart will not refresh. I [made PR to make it possible](https://github.com/filamentphp/filament/pull/2729), but it is waiting for approval.

**Reply by aldesrahim** [reply]
the `updateTableFiltersDateDateFrom()` is working fine. I've tested to `dd()` inside method inside widget class and it's not give me any outputs.

the current example:
widget class:
```
protected $listeners = ['halo'];

public function halo()
{
    dd('from halo');
}
```

listrecords class:
```
$this->emitTo('component-name', 'halo');
```

**Reply by studiowizjo** [reply]
So, `updateTableFiltersDateDateFrom()` is called, but `halo()` not?

**Reply by aldesrahim** [reply]
Yes, something like that

**Reply by studiowizjo** [reply]
This should work, at least it works for me. Have you tried emitting without scoping: `$this->emit('component-name', $data)`?

**Reply by aldesrahim** [reply]
I am sorry for late reply, It works when using `$this->emit('event-name')` without scoping to component using `$this->emitTo('component-name', 'event-name')`, I really dont know is this intended or not. Anyway, thanks for helping me with this !

**Reply by danharrin** [reply]
Tried `$this->emitTo(Widget::getName(), 'halo');` ?

**Reply by aldesrahim** [reply]
> Tried `$this->emitTo(Widget::getName(), 'halo');` ?

I've tried it and it is not working
